product-services/src/functions/create-product/createProduct.ts

```diff
@@ -0,0 +1,44 @@
+import "source-map-support/register";
+import {
+  ValidatedAPIGatewayProxyEvent,
+  formatJSONResponse,
+  ParsedAPIGatewayProxyEvent,
+} from "../../libs/apiGateway";
+import { middyfy } from "../../libs/lambda";
+import createHttpError from "http-errors";
+
+import { ProductServiceInstance } from "../../service/product-service";
+import { ProductServiceInterface } from "../../service/product-service-interface";
+import schema from "./schema";
+
+export type CreateProductEvent = ParsedAPIGatewayProxyEvent<typeof schema>;
+
+export class CreateProductController {
+  constructor(private service: ProductServiceInterface) {}
+
+  handler: ValidatedAPIGatewayProxyEvent<typeof schema> = async (
+    { body }: CreateProductEvent,
+    context
+  ) => {
+    console.log("Incoming request body", body);
+    context.callbackWaitsForEmptyEventLoop = false;
+    try {
+      const product = await this.service.createProduct(body);
+      if (product) {
+        return formatJSONResponse(200, product);
+      } else {
+        const response = new createHttpError.BadRequest(
+          `Product does not created`
+        );
+        return formatJSONResponse(response.statusCode, response.message);
+      }
+    } catch (error) {
+      const response = new createHttpError.InternalServerError(error.message);
+      return formatJSONResponse(response.statusCode, response.message);
+    }
+  };
+}
+
+export const handler = middyfy(
+  new CreateProductController(ProductServiceInstance).handler
+);
```

Comment:
U should add empty line at the end of each file

Reply:
Errors were fixed. Now adding a product works correctly.